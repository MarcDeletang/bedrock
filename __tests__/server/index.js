const importApp = require('../../app');
const importBase = require('../base.js');
const { Bedrock } = importApp;
const { rejectGenerator, handleResolveSnap, handleRejectSnap } = importBase;

const services = [];
const models = [];
const registers = [];
const decorations = [];
const routes = {};

describe('Testing the server initialisation', () => {

  it('Check server shape', () => {
    expect(Bedrock).toMatchSnapshot();
  });

  it('Reject with error (services)', () => {
    return handleRejectSnap(Bedrock.init);
  });

  it('Reject with error (models)', () => {
    return handleRejectSnap(Bedrock.init, services);
  });

  it('Reject with error (registers)', () => {
    return handleRejectSnap(Bedrock.init, services, models);
  });

    it('Reject with error (decorations)', () => {
    return handleRejectSnap(Bedrock.init, services, models, registers);
  });
  
  it('Reject with error (routes)', () => {
    return handleRejectSnap(Bedrock.init, services, models, registers, decorations);
  });

  it('Reject with error (env)', () => {
    delete process.env.NODE_ENV;
    return handleRejectSnap(Bedrock.init, {})
      .then(() => process.env.NODE_ENV = 'test');
  });

  it('Load without error', ()=> {
    return Bedrock.init(services, models, registers, decorations, routes, []).then(Bedrock=> expect(true).toBeTruthy());
  });

});
