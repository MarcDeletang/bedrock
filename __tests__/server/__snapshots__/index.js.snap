// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the server initialisation Check server shape 1`] = `
Object {
  "config": null,
  "env": "test",
  "hapi": null,
  "init": [Function],
  "log": MockLogger {},
}
`;

exports[`Testing the server initialisation Reject with error (decorations) 1`] = `[Error: decorations]`;

exports[`Testing the server initialisation Reject with error (env) 1`] = `[Error: process.env.NODE_ENV must be set]`;

exports[`Testing the server initialisation Reject with error (models) 1`] = `[Error: models]`;

exports[`Testing the server initialisation Reject with error (registers) 1`] = `[Error: registers]`;

exports[`Testing the server initialisation Reject with error (routes) 1`] = `[Error: routes]`;

exports[`Testing the server initialisation Reject with error (services) 1`] = `[Error: services]`;
